resource-resolver-plugins/StyleUrlResolvePlugin.js

```diff
@@ -0,0 +1,99 @@
+const ts = require("typescript");
+const fs = require("fs");
+const path = require("path");
+
+const StyleUrlResolvePlugin = (function() {
+    function StyleUrlResolvePlugin(options) {
+        if (!options || !options.platform) {
+            throw new Error(`Target platform must be specified!`);
+        }
+
+        this.platform = options.platform;
+    }
+
+    StyleUrlResolvePlugin.prototype.apply = function (compiler) {
+        compiler.plugin("make", (compilation, callback) => {
+            const aotPlugin = getAotPlugin(compilation);
+            aotPlugin._program.getSourceFiles()
+                .map(sf => this.setCurrentDirectory(sf))
```

Comment:
This one looks unnecessary. `usePlatformStyleUrl` does its own `setCurrentDirectory`